Docs-10014-PM2.6NewFeatures_ESB (#112)
mulesoft/docs-partner-manager: modules/ROOT/pages/endpoint-https-receive.adoc

|===
|Name |Description |Required
|Endpoint Name

> |The automatically generated name to identify the endpoint, which you can subsequently modify manually.

Reviewer: ditto re. periods: GLOBAL for list items

Reviewer 2: Periods added bc of mixed sentences/fragments

|Yes

|Basic authentication
a|When Basic authentication is selected, specify the user name and password to protect the HTTP listener.

After saving your new or reconfigured endpoint, you cannot subsequently retrieve or modify the password.

The backend applications that send JSON or XML payloads to the HTTP receive endpoint must pass the credentials in the HTTP request header.
|Yes

|API Key
a|The API key for the HTTP listener.

After selecting *API Key* in the *Authentication type* picklist, specify the HTTP header name and API key to protect the HTTP listener. After saving your new or reconfigured endpoint, you cannot subsequently retrieve or modify the API key.

The backend applications that send JSON or XML payloads to the HTTP receive endpoint must pass the API key in the HTTP request with the configured HTTP header name.
|Yes
|===

== HTTP or HTTPS Receive Endpoint URL

> === Deploy to CloudHub on Public Port

Reviewer: ```suggestion
=== Deploy to CloudHub on a Public Port
```

> When an HTTP or HTTPS receive endpoint is deployed with the private port configuration enabled, the endpoint application is deployed on the public HTTPS port (8082), with the traffic routed through the xref:runtime-manager::dedicated-load-balancer-tutorial#shared-load-balancers [shared load balancer]. The value in the URL field is automatically generated when you deploy a message flow to CloudHub that uses this endpoint.

Reviewer: ```suggestion
When an HTTP or HTTPS receive endpoint is deployed with the private port configuration enabled, the endpoint application is deployed on the public HTTPS port (8082), with the traffic routed through the xref:runtime-manager::dedicated-load-balancer-tutorial#shared-load-balancers [shared load balancer]. The value in the URL field is automatically generated when you deploy a message flow that uses this endpoint to CloudHub.
```

> You can view this URL by selecting the receiving endpoint from your deployed message flow or by clicking the endpoint name from the endpoints list. You can share this URL with the partners that would send HTTP or HTTPS messages to this endpoint.

Reviewer: ditto re. "would"

Reviewer 2: Done

> If your organization uses CloudHub VPC without a dedicated load balancer, your organization’s platform administrator can configure the necessary xref:runtime-manager::vpc-firewall-rules-concept.adoc[VPC Firewall Rules] to allow specific IP addresses of your

Reviewer: ```suggestion
If your organization uses CloudHub VPC without a dedicated load balancer, your organization’s platform administrator can configure the necessary xref:runtime-manager::vpc-firewall-rules-concept.adoc[VPC firewall rules] to allow specific IP addresses of your
```

partners to access the endpoint.

> === Deploy to CloudHub on Private Port

Reviewer: ```suggestion
=== Deploy to CloudHub on a Private Port
```

When an HTTP or HTTPS receive endpoint is deployed with the private port configuration enabled, the endpoint is deployed to the private HTTPS port (8092) and the internal URL of the endpoint appears on the the *Message Flow* or *Endpoint* detail page.

After deploying the first message flow using the HTTP or HTTPS receive endpoint configuration, you can view the runtime application name for the endpoint from the *Endpoint* details page or from the *Receive* section of your message flow. To configure xref:runtime-manager/lb-mapping-rules[URL mapping rules] in the xref:runtime-manager::cloudhub-dedicated-load-balancer.adoc[DLB] to forward HTTP or HTTPS messages received from your backend applications to outbound B2B message flows, provide the application name to your organization’s platform administrator.

If your backend processes consuming the HTTP or HTTPS receive endpoints are also running inside the same VPC, they can call the internal URL directly.

> If your backend processes consuming the endpoint are in a different network, your organization’s platform administrator can update DLB settings to xref:runtime-manager::lb-whitelists.adoc[whitelist] specific IP addresses to access the endpoint.

Reviewer: ditto comment Line 62

Reviewer 2: Done

Reviewer 2: Done

> After deploying the first message flow using the HTTP or HTTPS receive endpoint configuration, you can view the runtime application name for the endpoint from the *Endpoint* details page or from the *Receive* section of your message flow. To configure xref:runtime-manager/lb-mapping-rules[URL mapping rules] in the xref:runtime-manager::cloudhub-dedicated-load-balancer.adoc[Dedicated Load Balancer] to forward HTTP or HTTPS messages received from your backend applications to outbound B2B message flows, provide the application name to your organization’s platform administrator.

Reviewer: ```suggestion
After deploying the first message flow using the HTTP or HTTPS receive endpoint configuration, you can view the runtime application name for the endpoint from the *Endpoint* details page or from the *Receive* section of your message flow. To configure xref:runtime-manager/lb-mapping-rules[URL mapping rules] in the xref:runtime-manager::cloudhub-dedicated-load-balancer.adoc[dedicated load balancer] to forward HTTP or HTTPS messages received from your backend applications to outbound B2B message flows, provide the application name to your organization’s platform administrator.
```

The external URL that you can share with your partners is `https://{DLB-domain}/{Input-path}/{base-path}/{message-type}`.

For example, if the name of your DLB domain is `mythical.lb.anypointdns.net`, the runtime application name is `b2b-inbound-http-ognq`, and the base path you configured in the HTTPS receive endpoint is `send2`, then your administrator adds a URL mapping rule in the DLB settings to forward incoming requests to the runtime application within your VPC:

image::URL-mapping-rules-http.png[URL mapping rules in the DLB]

> With these configurations:

Reviewer: I missed this: we don't want to continue a text that introduces a graphic after the graphic as well. Reword to create a sentence intro.

Reviewer 2: Done

> If your backend processes consuming the endpoint are in a different network, your organization’s platform administrator can update Dedicated Load Balancer settings to xref:runtime-manager::lb-whitelists.adoc[whitelist] specific IP addresses to access the endpoint.

Reviewer: GLOBAL lc/abbr. DLB; I'll stop marking from here.

Reviewer 2: Done

> * The external HTTPS receive endpoint URL is `https://mythical.lb.anypointdns.net/apm/qa/outbound/send2/``.

Reviewer: Another double `...if this is intentional, then STET:
```suggestion
* The external HTTPS receive endpoint URL is `https://mythical.lb.anypointdns.net/apm/qa/outbound/send2/`.
```

> The external URL that you will share with your partners will be https://{DLB-domain}/{Input-path}/{base-path}/{message-type}.

Reviewer: Reword to present tense.
GLOBALLY search for "will" and try to reword, as possible.

Reviewer: GLOBALLY format URLs using mono tagging.

Reviewer 2: Done

> * Your backend applications send the application JSON/XML messages for your outbound message flows from `https://mythical.lb.anypointdns.net/apm/qa/outbound/send2/{message-type}``.

Reviewer: ```suggestion
* Your backend applications send the application JSON/XML messages for your outbound message flows from `https://mythical.lb.anypointdns.net/apm/qa/outbound/send2/{message-type}`.
```

> For example, if the name of your DLB domain is `mythical.lb.anypointdns.net`, and the runtime application name is `b2b-inbound-http-ognq`, and the base path you configured in the HTTPS receive endpoint is `send2`, then your administrator will add a URL mapping rule in the Dedicated Load Balancer settings to forward incoming requests to the runtime application within your VPC.

Reviewer: ```suggestion
For example, if the name of your DLB domain is `mythical.lb.anypointdns.net`, the runtime application name is `b2b-inbound-http-ognq`, and the base path you configured in the HTTPS receive endpoint is `send2`, then your administrator adds a URL mapping rule in the DLB settings to forward incoming requests to the runtime application within your VPC:
```

> 

Reviewer: URLs > ` `

Reviewer 2: Done

* If `enterprise-ob-po-ack` is the message type, your backend systems post the HTTPS requests to the URL
`https://mythical.lb.anypointdns.net/apm/qa/outbound/send2/enterprise-ob-po-ack`.

> === Deploy to On-Premises Mule Instance

Reviewer: ```suggestion
=== Deploy to an On-Premises Mule Instance
```

> === Deploy to On-Premise Mule Runtime

Reviewer: ```suggestion
=== Deploy to On-Premises Mule Instance
```

> If you are deploying to an on-premises Mule instance with a firewall and load balancer, work with your hosting or network security team to obtain the externally accessible DNS or name of the host where the Mule server is running, and replace `{runtime_host}` in the URL with the DNS name of your host. For example, if your DNS host name is `b2b.mycompany.com`, the receiver URL is `+https://b2b.mycompany.com/base-path/+`, and your backend systems should be posting the HTTPS request to https://b2b.mycompany.com/base-path/{message-type}.

Reviewer: Mule server? Do you mean Mule runtime engine (Mule)?

Reviewer: Also, "For example, if your DNS host name is `b2b.mycompany.com`, the receiver URL is `+https://b2b.mycompany.com/base-path/+` and your backend systems post the HTTPS request to `https://b2b.mycompany.com/base-path/{message-type}`.
```

Reviewer 2: Done
